libs/cookies/src/lib/directives/has-cookie/has-cookie.directive.ts

```diff
@@ -0,0 +1,175 @@
+import {
+	ChangeDetectorRef,
+	ComponentRef,
+	Directive,
+	EmbeddedViewRef,
+	Inject,
+	Input,
+	Optional,
+	TemplateRef,
+	Type,
+	ViewContainerRef,
+} from '@angular/core';
+import { Subject, tap, takeUntil, combineLatest, map } from 'rxjs';
+import { flatten } from 'lodash';
+
+import { NgxHasCookieConfiguration } from '../../types';
+import { NgxCookieService } from '../../services';
+import { NgxCookiesFallBackComponent } from '../../abstracts';
+import { NgxCookiesFallbackComponentToken } from '../../tokens';
+
+/**
+ * A structural directive that provides a way to render UI elements based on whether a (set of) cookie(s) have been accepted by the user.
+ */
+@Directive({
+	selector: '[hasCookie]',
+	standalone: true,
+})
+export class NgxHasCookieDirective {
+	/**
+	 * The destroyed state of the directive
+	 */
+	private destroyed$: Subject<void>;
```

Comment:
How do we feel about using our own packages within each other? Because we could use the DestroyComponent here 🤔

Reply:
It's something we have to do at one point; but we currently don't have a setup for the shared lib yet 😂 I didn't want to do it in this PR as it's already quite big, but I'll add a TODO comment!

Reply:
That's fair. Let's keep this on the radar though? Maybe make a ticket on the project board?